analysis/heatmapLSF.py

```diff
@@ -1,89 +1,155 @@
-from scripts.loading_utils import load_config, load_model
-from scripts.data.custom import CustomTest as CustomDataset
-from scripts.models.LSFautoencoder import LSFVQVAE
+import argparse, os, sys, datetime, glob, importlib
+from omegaconf import OmegaConf
+import numpy as np
+import torch
+from torch.utils.data import DataLoader, Dataset
+import pytorch_lightning as pl
+from pytorch_lightning.trainer import Trainer
+
+from tqdm import tqdm
+from scripts.data.utils import custom_collate
 from scripts.plotting_utils import get_fig_pth
 from scripts.analysis_utils import aggregate_metric_from_specimen_to_species, get_CosineDistance_matrix
 from scripts.plotting_utils import plot_heatmap
 
-import torch
-import os
-import numpy as np
-from omegaconf import OmegaConf
-import argparse
+def get_obj_from_str(string, reload=False):
+    module, cls = string.rsplit(".", 1)
+    if reload:
+        module_imp = importlib.import_module(module)
+        importlib.reload(module_imp)
+    return getattr(importlib.import_module(module, package=None), cls)
+
+
+def get_parser(**parser_kwargs):
+    parser = argparse.ArgumentParser(**parser_kwargs)
+    parser.add_argument(
+        "-b",
+        "--config",
+        default="configs/lsf_inference.yaml",
+    )
+    return parser
 
 
+def instantiate_from_config(config):
+    if not "target" in config:
+        raise KeyError("Expected key `target` to instantiate.")
+    return get_obj_from_str(config["target"])(**config.get("params", dict()))
 
 
-def save_cosine_distance_matrix(model, dataset, idx_to_label, plot_save_dir, device=None):
-        encodings = []
-        classes = []
-        with torch.no_grad():
-            for d in dataset:
-                img = torch.tensor(d['image']).unsqueeze(0).permute(0, 3, 1, 2)
-                if device is not None:
-                    img = img.cuda()
-                z, _, _ = model.image2encoding(img)
-                z = (z @ model.LSF_disentangler.M.t())[:, :38]
-                encodings.append(z)
-                classes.append(d['class'])
+class WrappedDataset(Dataset):
+    """Wraps an arbitrary object with __len__ and __getitem__ into a pytorch dataset"""
+    def __init__(self, dataset):
+        self.data = dataset
 
-        encodings = torch.cat(encodings)
+    def __len__(self):
+        return len(self.data)
 
-        sort_indices = np.argsort(classes)
-        sorted_encodings = encodings[sort_indices,:]
-        sorted_classes = sorted(classes)
+    def __getitem__(self, idx):
+        return self.data[idx]
 
-        z_cosine_distances = get_CosineDistance_matrix(sorted_encodings)
 
-        os.makedirs(plot_save_dir, exist_ok=True)
+class DataModuleFromConfig(pl.LightningDataModule):
```

Comment:
This function already exists in main. cant we just import it?

Reply:
same with other classes in this file

Reply:
Modified Moe